lib/ansible/modules/cloud/cloudstack/cs_instance.py

```diff
@@ -77,11 +77,12 @@
       - Name the hypervisor to be used for creating the new instance.
       - Relevant when using I(state=present), but only considered if not set on ISO/template.
       - If not set or found on ISO/template, first found hypervisor will be used.
-    choices: [ KVM, VMware, BareMetal, XenServer, LXC, HyperV, UCS, OVM, Simulator ]
+    choices: [KVM, kvm, VMware, vmware, BareMetal, baremetal, XenServer, xenserver, LXC, lxc, HyperV, hyperv, UCS, ucs, OVM, ovm, Simulator, simulator]
   keyboard:
     description:
       - Keyboard device type for the instance.
     choices: [ de, de-ch, es, fi, fr, fr-be, fr-ch, is, it, jp, nl-be, no, pt, uk, us ]
```

Comment:
Quote the `no` to avoid parsing as a bool.

Reply:
*facepalm*